src/lib.rs

```diff
@@ -309,24 +313,23 @@ impl<F: io::Write> io::Write for File<F> {
 #[derive(Debug, Clone, Copy)]
 pub struct DelimCodec<D>(pub D);
 
-impl<D: Into<u8> + Copy> tokio_core::io::Codec for DelimCodec<D> {
-    type In = Vec<u8>;
-    type Out = Vec<u8>;
+impl<D: Into<u8> + Copy> codec::Decoder for DelimCodec<D> {
+    type Item = Vec<u8>;
+    type Error = io::Error;
 
-    fn decode(&mut self, buf: &mut tokio_core::io::EasyBuf)
-              -> io::Result<Option<Self::In>> {
+    fn decode(&mut self, buf: &mut BytesMut) -> io::Result<Option<Self::Item>> {
         Ok(buf.as_ref().iter().position(|b| *b == self.0.into())
-           .map(|n| buf.drain_to(n + 1).as_ref().to_vec()))
+           .map(|n| buf.split_to(n + 1).as_ref().to_vec()))
     }
+}
```

Comment:
I think `decode_eof` still has to be implemented. The default implementation fails if there is data left in the buffer.